### PR TITLE
@ #17 | should install ansible through pip in guest mode

### PR DESCRIPTION
**Pull request by hieptranquoc**
connect #17
related https://github.com/teracyhq-incubator/teracy-dev-entry-k8s/issues/10

### CODE REVIEW

**Comment by hieptranquoc** [comment]
please check this PR! Thank you!

**Comment by hieptranquoc** [comment]
@hoatle I've updated. Please check this PR again. Thank you!

File: config.yaml [hunk 1]
```diff
@@ -2,7 +2,10 @@
 teracy-dev-certs:
   # the node id which certs will provision
   node_id: "0" # 0 by default from teracy-dev-core
-  ansible_mode: guest # or host to run ansible from the host machine
```
Comment (hoatle) [inline config.yaml]: if ansible_mode exists => add deprecated warnings and set that value to ansible.mode

File: lib/teracy-dev-certs/processors/certs.rb [hunk 2]
```diff
@@ -34,21 +34,40 @@ def setup(certs_config)
 
       def generate_node(certs_config)
         ansible_type = "ansible_local" # guest by default
-        ansible_type = "ansible" if certs_config['ansible_mode'] == 'host'
```
Comment (hoatle) [inline lib/teracy-dev-certs/processors/certs.rb]: add deprecated warning instead of removing, remember to always handle backward compatibility

**Comment by hieptranquoc** [comment]
@hoatle I've updated. Please check this PR again.!

File: lib/teracy-dev-certs/processors/certs.rb [hunk 3]
```diff
@@ -33,22 +33,46 @@ def setup(certs_config)
...
           "common_name" => certs_config['common_name'],
           "alt_names" => certs_config['alt_names'],
           "ca_days" => certs_config['ca_days'],
           "cert_days" => certs_config['cert_days'],
           "certs_path" => '/vagrant/workspace/certs'
         }
+
+        if certs_config['ansible']['mode'] == 'guest'
+          ansible_install_mode = certs_config['ansible']['install_mode']
+          ansible_install_mode = 'pip' unless TeracyDev::Util.exist?(certs_config['ansible']['install_mode'])
```
Comment (hoatle) [inline lib/teracy-dev-certs/processors/certs.rb]: set the default "pip" for yaml config instead (to fix https://github.com/teracyhq-incubator/teracy-dev-certs/issues/12)

File: lib/teracy-dev-certs/processors/certs.rb [hunk 4]
```diff
@@ -33,22 +33,46 @@ def setup(certs_config)
...
+
+          ansible_install_version = certs_config['ansible']['version']
+          ansible_install_version = 'latest' unless TeracyDev::Util.exist?(certs_config['ansible']['version'])
+
+          provisioner = {
+            "_id" => "certs-ansible",
+            "type" => ansible_type,
+            "extra_vars" => extra_vars,
+            "install_mode" => ansible_install_mode,
```
Comment (hoatle) [inline lib/teracy-dev-certs/processors/certs.rb]: set install_mode if ansible_install_mode is valid (not empty)

File: lib/teracy-dev-certs/processors/certs.rb [hunk 5]
```diff
@@ -33,22 +33,46 @@ def setup(certs_config)
...
           "cert_days" => certs_config['cert_days'],
           "certs_path" => '/vagrant/workspace/certs'
         }
+
+        if certs_config['ansible']['mode'] == 'guest'
+          ansible_install_mode = certs_config['ansible']['install_mode']
+          ansible_install_mode = 'pip' unless TeracyDev::Util.exist?(certs_config['ansible']['install_mode'])
+
+          ansible_install_version = certs_config['ansible']['version']
+          ansible_install_version = 'latest' unless TeracyDev::Util.exist?(certs_config['ansible']['version'])
```
Comment (hoatle) [inline lib/teracy-dev-certs/processors/certs.rb]: don't set "latest" as default (similar to the related issue on teracy-dev-k8s)

File: lib/teracy-dev-certs/processors/certs.rb [hunk 6]
```diff
@@ -33,22 +33,46 @@ def setup(certs_config)
...
+
+          provisioner = {
+            "_id" => "certs-ansible",
+            "type" => ansible_type,
+            "extra_vars" => extra_vars,
+            "install_mode" => ansible_install_mode,
+            "version" => ansible_install_version
```
Comment (hoatle) [inline lib/teracy-dev-certs/processors/certs.rb]: set this only if ansible_install_version is valid (not empty)

**Comment by hoatle** [comment]
merged, I'm going to add more improvements later on.

**Pull request merged**